Sources/Commands/SwiftPackageRegistryTool.swift

```diff
@@ -0,0 +1,185 @@
+            // TODO: Require login if password is specified
+
+            let path = try swiftTool.getRegistryConfigurationPath(fileSystem: localFileSystem, global: global)
+
+            var configuration = RegistryConfiguration()
+            if localFileSystem.exists(path) {
+                configuration = try RegistryConfiguration.readFromJSONFile(in: localFileSystem, at: path)
+            }
+
+            if let scope = scope {
+                configuration.scopedRegistries[scope] = .init(url: url)
+            } else {
+                configuration.defaultRegistry = .init(url: url)
+            }
+
+            try configuration.writeToJSONFile(in: localFileSystem, at: path)
+
+            // TODO: Add login and password to .netrc
+        }
+    }
+
+    struct Unset: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Remove a configured registry")
+
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+
+        @Flag(help: "Apply settings to all projects for this user")
+        var global: Bool = false
+
+        @Option(help: "Associate the registry with a given scope")
+        var scope: String?
+
+        func run(_ swiftTool: SwiftTool) throws {
+            let path = try swiftTool.getRegistryConfigurationPath(fileSystem: localFileSystem, global: global)
+            var configuration = try RegistryConfiguration.readFromJSONFile(in: localFileSystem, at: path)
+
+            if let scope = scope {
+                guard let _ = configuration.scopedRegistries[scope] else {
+                    throw RegistryConfigurationError.missingScope(scope)
+                }
+                configuration.scopedRegistries.removeValue(forKey: scope)
+            } else {
+                guard let _ = configuration.defaultRegistry else {
+                    throw RegistryConfigurationError.missingScope()
+                }
+                configuration.defaultRegistry = nil
+            }
+
+            try configuration.writeToJSONFile(in: localFileSystem, at: path)
+        }
+    }
+}
+
+// MARK: -
+
+private extension Decodable {
+    static func readFromJSONFile(in fileSystem: FileSystem, at path: AbsolutePath) throws -> Self {
+        let content = try fileSystem.readFileContents(path)
+
+        let decoder = JSONDecoder()
+
+        return try decoder.decode(Self.self, from: Data(content.contents))
+    }
+}
+
+private extension Encodable {
+    func writeToJSONFile(in fileSystem: FileSystem, at path: AbsolutePath) throws {
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = [.sortedKeys, .prettyPrinted, .withoutEscapingSlashes]
+
+        let data = try encoder.encode(self)
+
+        try fileSystem.writeFileContents(path, bytes: ByteString(data), atomically: true)
+    }
+}
+
+private extension SwiftTool {
+    func getRegistryConfigurationPath(fileSystem: FileSystem = localFileSystem, global: Bool) throws -> AbsolutePath {
```

Comment:
I would remove the default fileSystem unless there is a strong reason to set it? its something we need to do all over the code base, but might as well not add to this anti-pattern

Reply:
Totally agree. Only reason for this was to match existing usage.